jax/_src/lax/parallel.py

```diff
@@ -730,6 +740,82 @@ def pos_reduce(x):
 core.axis_substitution_rules[pmin_p] = partial(_subst_all_names_in_param, 'axes')
 
 
+# TODO(mattjj): merge with lax.argmin_p / lax.argmax_p
+pargmax_p = core.Primitive('pargmax')
+pargmin_p = core.Primitive('pargmin')
+
+def _pargminmax_impl(prim, x, *, axis):
+  if not isinstance(axis, int): raise NameError(f"unbound axis name: {axis}")
+  return xla.apply_primitive(prim, x, axis=axis)
+pargmax_p.def_impl(partial(_pargminmax_impl, pargmax_p))
+pargmin_p.def_impl(partial(_pargminmax_impl, pargmin_p))
+
+def _pargminmax_abstract_eval(x, *, axis):
+  if isinstance(axis, int):
+    if not 0 <= axis < np.ndim(x):
+      raise ValueError(f"axis index {axis} for operand shape {np.shape(x)}")
+    shape = tuple(np.delete(np.shape(x), axis))
+  else:
+    # TODO(mattjj): avals with names, check and eliminate axis name
+    shape = np.shape(x)
+  return ShapedArray(shape, np.dtype('int32'))
+pargmax_p.def_abstract_eval(_pargminmax_abstract_eval)
+pargmin_p.def_abstract_eval(_pargminmax_abstract_eval)
+
+def _pargminmax_vmap_collective_rule(reducer, frame, vals_in, dims_in, *, axis):
+  del frame, axis  # Unused.
+  (x,), (d,) = vals_in, dims_in
+  return reducer(x, d, np.dtype('int32')), None
```

Comment:
I think that you don't have a guarantee that `d` is not `batching.not_mapped` here, so we might have to handle that case explicitly.

Reply:
I think that should be guaranteed by [this code](https://github.com/google/jax/blob/5c14dafefef1ea891bab4fafd2d8d61afa927e8e/jax/interpreters/batching.py#L153-L154).